diff --git a/CMakeLists.txt b/CMakeLists.txt
index 01f62a1a8..71db29e2f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -86,7 +86,7 @@ target_link_libraries(
 
 cmake_policy(SET CMP0063 NEW)
 set(CMAKE_CXX_EXTENSIONS OFF)
-set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_VISIBILITY_PRESET hidden)
 set(CMAKE_C_EXTENSIONS OFF)
diff --git a/platform/qt/src/mbgl/http_file_source.cpp b/platform/qt/src/mbgl/http_file_source.cpp
index a2a13c5e1..a4ac0c58e 100644
--- a/platform/qt/src/mbgl/http_file_source.cpp
+++ b/platform/qt/src/mbgl/http_file_source.cpp
@@ -45,7 +45,8 @@ void HTTPFileSource::Impl::request(HTTPRequest* req)
 #if QT_VERSION >= QT_VERSION_CHECK(5, 15, 0)
     connect(data.first, &QNetworkReply::errorOccurred, this, &HTTPFileSource::Impl::onReplyFinished);
 #else
-    connect(data.first, &QNetworkReply::error, this, &HTTPFileSource::Impl::onReplyFinished);
+    connect(data.first, SIGNAL(error()), this, SLOT(onReplyFinished()));
+    //connect(data.first, &QNetworkReply::error, this, &HTTPFileSource::Impl::onReplyFinished);
 #endif
 }
 
