diff --git a/platform/qt/src/mbgl/run_loop.cpp b/platform/qt/src/mbgl/run_loop.cpp
index 6bd9cb035c5..aef0d2b98da 100644
--- a/platform/qt/src/mbgl/run_loop.cpp
+++ b/platform/qt/src/mbgl/run_loop.cpp
@@ -96,7 +96,7 @@ void RunLoop::addWatch(int fd, Event event, std::function<void(int, Event)>&& cb
     if (event == Event::Read || event == Event::ReadWrite) {
         auto notifier = std::make_unique<QSocketNotifier>(fd, QSocketNotifier::Read);
         QObject::connect(notifier.get(), &QSocketNotifier::activated, impl.get(), &RunLoop::Impl::onReadEvent);
-        impl->readPoll[fd] = WatchPair(std::move(notifier), std::move(cb));
+        impl->readPoll[fd] = WatchPair(std::move(notifier), cb);
     }
 
     if (event == Event::Write || event == Event::ReadWrite) {
